optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/employee/EmployeeRestServiceImpl.java

```diff
@@ -94,17 +96,26 @@ protected void validateTenantIdParameter(Integer tenantId, Employee employee) {
 
     @Override
     @Transactional
-    public Long addEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
+    public EmployeeAvailabilityView addEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
         EmployeeAvailability employeeAvailability = convertFromView(tenantId, employeeAvailabilityView);
         entityManager.persist(employeeAvailability);
-        return employeeAvailability.getId();
+        return new EmployeeAvailabilityView(employeeAvailability);
     }
 
     @Override
     @Transactional
-    public void updateEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
+    public EmployeeAvailabilityView updateEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
         EmployeeAvailability employeeAvailability = convertFromView(tenantId, employeeAvailabilityView);
         entityManager.merge(employeeAvailability);
+        // employeeAvailability is detached, cannot get version from it
+        entityManager.flush();
+        return getEmployeeAvailability(tenantId, employeeAvailability.getId());
```

Comment:
the flush and the getEA() should be obsolete if you have the return value of the merge, just return new EAView(employeeAvailability)

Reply:
Nope, it was that before, version was not incremented, which caused issues

Reply:
@Christopher-Chianelli See the description [here](https://stackoverflow.com/a/1070629) of the difference between persist and merge. After a merge call the parameter is not managed, but the return value is a JPA managed instance that should contain all updated fields.